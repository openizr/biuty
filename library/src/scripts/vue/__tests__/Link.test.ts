/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { mount } from '@vue/test-utils';
import UILink from 'scripts/vue/Link.vue';

describe('vue/UILink', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const wrapper = mount(UILink, {
      propsData: { label: 'Test', href: 'https://test.com', modifiers: 'large' },
    });
    wrapper.trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    const wrapper = mount(UILink, {
      propsData: { label: 'Test', href: 'https://test.com', id: 'test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders correctly - with listener', () => {
    const onClick = jest.fn();
    const wrapper = mount(UILink, {
      propsData: { label: 'Test', href: 'https://test.com', onClick },
    });
    wrapper.trigger('click');
    expect(wrapper.html()).toMatchSnapshot();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
