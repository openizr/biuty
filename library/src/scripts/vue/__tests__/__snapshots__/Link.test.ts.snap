// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue/UILink > renders correctly - basic 1`] = `
<a
  class="ui-link ui-link--large"
  href="https://test.com"
>
  Test
</a>
`;

exports[`vue/UILink > renders correctly - with id 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  id="test"
>
  Test
</a>
`;

exports[`vue/UILink > renders correctly - with rel 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  rel="no referrer"
>
  Test
</a>
`;

exports[`vue/UILink > renders correctly - with target 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  target="_blank"
>
  Test
</a>
`;

exports[`vue/UILink renders correctly - basic 1`] = `
<a
  class="ui-link ui-link--large"
  href="https://test.com"
>
  Test
</a>
`;

exports[`vue/UILink renders correctly - with id 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  id="test"
>
  Test
</a>
`;

exports[`vue/UILink renders correctly - with rel 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  rel="no referrer"
>
  Test
</a>
`;

exports[`vue/UILink renders correctly - with target 1`] = `
<a
  class="ui-link"
  href="https://test.com"
  target="_blank"
>
  Test
</a>
`;
